# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1380772348 14400
#      Wed Oct 02 23:52:28 2013 -0400
# Node ID a646665cd57409462e502f39bf58b7525876e501
# Parent  92541ff4cc3c3fcafb8935b6f34fb4c816c27622
eliminate unecessary static function in urlwrite.cc

* urlwrite.cc (cleanup_urlwrite): Delete.
(Furlwrite): Use octave_unlink instead of cleanup_urlwrite.

diff --git a/libinterp/dldfcn/urlwrite.cc b/libinterp/dldfcn/urlwrite.cc
--- a/libinterp/dldfcn/urlwrite.cc
+++ b/libinterp/dldfcn/urlwrite.cc
@@ -823,32 +823,27 @@ ch_manager::do_free (const curl_handle& 
       else
         error ("ch_manager::free: invalid object %g", h.value ());
     }
 }
 
 ch_manager *ch_manager::instance = 0;
 
 static void
-cleanup_urlwrite (std::string filename)
-{
-  octave_unlink (filename);
-}
-
-static void
 reset_path (const curl_object& curl)
 {
   curl.cwd ("..");
 }
 
 static void
 delete_file (std::string file)
 {
   octave_unlink (file);
 }
+
 #endif
 
 DEFUN_DLD (urlwrite, args, nargout,
   "-*- texinfo -*-\n\
 @deftypefn  {Loadable Function} {} urlwrite (@var{url}, @var{localfile})\n\
 @deftypefnx {Loadable Function} {@var{f} =} urlwrite (@var{url}, @var{localfile})\n\
 @deftypefnx {Loadable Function} {[@var{f}, @var{success}] =} urlwrite (@var{url}, @var{localfile})\n\
 @deftypefnx {Loadable Function} {[@var{f}, @var{success}, @var{message}] =} urlwrite (@var{url}, @var{localfile})\n\
@@ -969,17 +964,17 @@ urlwrite (\"http://www.google.com/search
   if (! ofile.is_open ())
     {
       error ("urlwrite: unable to open file");
       return retval;
     }
 
   unwind_protect_safe frame;
 
-  frame.add_fcn (cleanup_urlwrite, filename);
+  frame.add_fcn (delete_file, filename);
 
   bool ok;
   curl_object curl = curl_object (url, method, param, ofile, ok);
 
   ofile.close ();
 
   if (!error_state)
     frame.discard ();
